src/MetaMorpho.sol

```diff
@@ -520,15 +521,23 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
     function deposit(uint256 assets, address receiver) public override returns (uint256 shares) {
         uint256 newTotalAssets = _accrueFee();
 
+        // `newTotalAssets + assets` may be a little off from `totalAssets()`.
+        _updateLastTotalAssets(newTotalAssets + assets);
```

Comment:
```suggestion
        lastTotalAssets = newTotalAssets;
```

Reply:
1. Otherwise we emit the event twice
2. At this stage, only newTotalAssets are on the vault